### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by ArtDu**
* Fix logic for determinism for asynchronous tests
* Always failing tests have been fixed

A detailed description can be found in the commit message

### CODE REVIEW

File: src/main/java/io/tarantool/driver/core/TarantoolConnectionImpl.java [hunk 1]
```diff
@@ -66,6 +70,8 @@ public boolean isConnected() {
             if (!f.isSuccess()) {
                 requestFuture.completeExceptionally(
                         new RuntimeException("Failed to send the request to Tarantool server", f.cause()));
+            } else {
+                logger.info("Request {} sent, status Success", request.getHeader().getSync());
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/core/TarantoolConnectionImpl.java:74]: Change info to debug

**Review by akudiyar — CHANGES_REQUESTED** [review]

File: src/test/java/io/tarantool/driver/integration/ClusterConnectionIT.java [hunk 2]
```diff
@@ -73,7 +75,8 @@ private RetryingTarantoolTupleClient setupClusterClient(TarantoolClientConfig co
         ProxyTarantoolTupleClient client = new ProxyTarantoolTupleClient(clusterClient);
         return new RetryingTarantoolTupleClient(client,
                 TarantoolRequestRetryPolicies.byNumberOfAttempts(retries, e -> {
-                    logger.info("Retrying exception {}", e.getMessage());
+                    logger.debug("Retrying exception", e);
```
Comment (akudiyar) [inline src/test/java/io/tarantool/driver/integration/ClusterConnectionIT.java:78]: Debug logging

File: src/test/java/io/tarantool/driver/integration/ClusterConnectionIT.java [hunk 3]
```diff
@@ -105,49 +108,72 @@ void testMultipleRoutersReconnect_retryableRequestShouldNotFail() throws Excepti
...
-                prepareConfig().withConnectTimeout(1000).build(), addressProvider, 10, 0);
+        AtomicReference<Container.ExecResult> result = new AtomicReference<>();
+        AtomicReference<String> nextRouterName = new AtomicReference<>();
+
+        ClusterTarantoolTupleClient clusterClient =
+                new ClusterTarantoolTupleClient(prepareConfig().withConnectTimeout(1000).build(), addressProvider);
+        ProxyTarantoolTupleClient proxyClient = new ProxyTarantoolTupleClient(clusterClient);
+        RetryingTarantoolTupleClient client = new RetryingTarantoolTupleClient(proxyClient,
+                TarantoolRequestRetryPolicies.byNumberOfAttempts(10, e -> {
+                    logger.debug("Retrying exception ", e);
```
Comment (akudiyar) [inline src/test/java/io/tarantool/driver/integration/ClusterConnectionIT.java:119]: Debug logging

File: src/test/java/io/tarantool/driver/integration/ClusterConnectionIT.java [hunk 4]
```diff
@@ -105,49 +108,72 @@ void testMultipleRoutersReconnect_retryableRequestShouldNotFail() throws Excepti
...
+        assertEquals(0, result.get().getExitCode(), result.get().getStderr());
+        result.set(container.execInContainer(
+                "cartridge", "start", "--run-dir=/tmp/run", "--data-dir=/tmp/data", "-d", "router"));
+        assertEquals(0, result.get().getExitCode(), result.get().getStderr());
+        result.set(container.execInContainer(
+                "cartridge", "start", "--run-dir=/tmp/run", "--data-dir=/tmp/data", "-d", "second-router"));
+        assertEquals(0, result.get().getExitCode(), result.get().getStderr());
+
+        // wait until full reconnection completed
+        Thread.sleep(5000);
```
Comment (akudiyar) [inline src/test/java/io/tarantool/driver/integration/ClusterConnectionIT.java:169]: Can we wait for a line in logs here?

**Review by akudiyar — CHANGES_REQUESTED** [review]

File: src/main/java/io/tarantool/driver/AbstractTarantoolClient.java [hunk 5]
```diff
@@ -363,9 +363,11 @@ public TarantoolMetadataProvider metadataProvider() {
     @Override
     public <T, R extends List<T>> CompletableFuture<R> callForMultiResult(
             String functionName,
-            List<?> arguments, Supplier<R> resultContainerSupplier,
+            List<?> arguments,
+            Supplier<R> resultContainerSupplier,
             ValueConverter<Value, T> valueConverter) throws TarantoolClientException {
-        return callForMultiResult(functionName, arguments, resultContainerSupplier, valueConverter);
+        return callForMultiResult(functionName, arguments, config.getMessagePackMapper()
+                , resultContainerSupplier, valueConverter);
```
[the remark this answers is not in the document]
Comment (ArtDu) [reply src/main/java/io/tarantool/driver/AbstractTarantoolClient.java:370]: move the comma to the line above? how is done here? https://github.com/tarantool/cartridge-java/blob/1496c1425f834c997422864c055cfb9d7533d478/src/main/java/io/tarantool/driver/AbstractTarantoolClient.java#L411-L412

File: src/test/java/io/tarantool/driver/integration/ClusterConnectionIT.java (hunk 4 above)
Comment (ArtDu) [reply src/test/java/io/tarantool/driver/integration/ClusterConnectionIT.java]: By agreement, expectations were implemented through cartridge.is_healthy

File: src/main/java/io/tarantool/driver/AbstractTarantoolClient.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (akudiyar) [reply src/main/java/io/tarantool/driver/AbstractTarantoolClient.java]: Yes, commas shouldn't start the line

[the remark this answers is not in the document]
Comment (ArtDu) [reply src/main/java/io/tarantool/driver/AbstractTarantoolClient.java]: Then I will close the conversion since I have already fixed it

**Review by akudiyar — APPROVED** [review]

**Pull request merged**